crates/router/src/connector/bluesnap.rs

```diff
@@ -1095,31 +1095,36 @@ impl services::ConnectorRedirectResponse for Bluesnap {
         &self,
         _query_params: &str,
         json_payload: Option<serde_json::Value>,
-        _action: services::PaymentAction,
+        action: services::PaymentAction,
     ) -> CustomResult<payments::CallConnectorAction, errors::ConnectorError> {
-        let redirection_response: bluesnap::BluesnapRedirectionResponse = json_payload
-            .ok_or(errors::ConnectorError::MissingConnectorRedirectionPayload {
-                field_name: "json_payload",
-            })?
-            .parse_value("BluesnapRedirectionResponse")
-            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        match action {
+            services::PaymentAction::PSync => Ok(payments::CallConnectorAction::Trigger),
+            services::PaymentAction::CompleteAuthorize => {
+                let redirection_response: bluesnap::BluesnapRedirectionResponse = json_payload
```

Comment:
If expected json_payload is not present then we should return payments::CallConnectorAction::Avoid, since we know our script always have that json_payload, Let's take this in separate PR